UnitTests/CommonTestUtils/ConfigureJoinableTaskFactoryAttribute.cs

```diff
@@ -14,13 +16,25 @@ namespace CommonTestUtils
     [AttributeUsage(AttributeTargets.Assembly)]
     public sealed class ConfigureJoinableTaskFactoryAttribute : Attribute, ITestAction
     {
+        private readonly bool _checkForUpdatesOriginalValue = AppSettings.CheckForUpdates;
```

Comment:
💡 I wouldn't bother with trying to restore the value.

Reply:
The tests shall not bollix my `AppSettings`. I am afraid they will be stored to `%APPDATA%`.

Reply:
It's not safe to assume the cleanup code will execute. If there is a concern with storage in APPDATA, it needs to be addressed separately. I'll look into this.

Reply:
A new attribute should be added that implements `ITestAction` and applied to the assembly. This attribute should set the value of `AppSettings.SettingsContainer` to a transient container which ignores the setting in APPDATA and also does not persist any changes there. Before each test, the container should be reset to its default values.

Reply:
Thank you.

Reply:
Moved to #7654.